parsons/google/google_bigquery.py

```diff
@@ -1267,6 +1295,38 @@ def _fetch_query_results(self, cursor) -> Table:
         ptable = petl.frompickle(temp_filename)
         return Table(ptable)
 
+    def _validate_copy_inputs(self, if_exists: str, data_type: str):
+        if if_exists not in ["fail", "truncate", "append", "drop"]:
+            raise ValueError(
+                f"Unexpected value for if_exists: {if_exists}, must be one of "
+                '"append", "drop", "truncate", or "fail"'
+            )
+        if data_type not in ["csv", "json"]:
+            raise ValueError(
+                f"Only supports csv or json files [data_type = {data_type}]"
+            )
+
+    def _load_table_from_uri(self, source_uris, destination, job_config, **load_kwargs):
+        try:
+            load_job = self.client.load_table_from_uri(
+                source_uris=source_uris,
+                destination=destination,
+                job_config=job_config,
+                **load_kwargs,
+            )
+            load_job.result()
+            return load_job
+        except exceptions.BadRequest as e:
+            for idx, error_ in enumerate(load_job.errors):
+                if idx == 0:
+                    logger.error(
+                        "* Load job failed. Enumerating errors collection below:"
+                    )
+                logger.error(f"** Error collection - index {idx}:")
+                logger.error(error_)
+
+            raise e
```

Comment:
moved this logic to loop through the error collections into the lower level function because I think we'll always want to do this. @IanRFerguson I know we had that weird edge case where some of this logic was throwing. Do you remember what that was? (should have made a unit test for it)

Reply:
yeah that was when the Catalist flat file size exceeds Google's maximum compressed file size ... any of the state-level models are a good test case for that